forum/qa-plugin/report-reason/script.js

```diff
@@ -0,0 +1,116 @@
+        }
+    }
+    function showError(errorText) {
+        const paragraph = document.createElement("p");
+        paragraph.textContent = errorText;
+                    
+        removeAllChildFromErrorPopup();
+        errorPopup.appendChild(paragraph);
+        errorPopup.hidden = false
+    }
+
+    wrap.addEventListener("click", (e) => {
+        e.stopPropagation();
+    });
+    
+    commentSubmitButton.addEventListener("click", () => {
+       location.reload(); 
+    });
+    
+    body.addEventListener("click", (event) => {
+        if (event.target.matches(".qa-form-light-button-unflag")) {
+            location.reload();
+        }
+        if (event.target.matches(".qa-form-light-button-clearflags")) {
+            location.reload();
+        }
+        if (event.target.matches("input.qa-form-light-button-flag")) {
+            const flagButton = event.target;
+            flagButton.type = "button";
+            
+            event.preventDefault();
+            showPopup();    
+
+            flagboxPopup.addEventListener("click", () => {
+                hidePopup();
+            });
+
+            closer.addEventListener("click", () => {
+                hidePopup();
+                return false;
+            });
+
+            sendButton.addEventListener("click", () => {
+
+                const flagReason = document.querySelector("input.qa-spam-reason-radio:checked");
+                const flagNotice = document.querySelector(".qa-spam-reason-text").value;
+
+                if (!flagReason) {
+                    showError(reportReasonEmptyError);
+                    return false;
+                }
+
+                const { postid: postId, posttype: postType, parentid: parentId } = flagButton.dataset;
+                const dataArray = {questionid: flagQuestionid, postid: postId, posttype: postType, reasonid: flagReason.value, notice: flagNotice};
+                const sendData = JSON.stringify(dataArray);
+                let isError = false;
+
+                if(flagReason) {
+                    sendButton.value = "Wysyłanie...";
+                    sendButton.disabled = true;
+                    
+                    $.ajax({
+                        type: "POST",
+                        url: flagAjaxURL,
+                        data: { ajaxdata: sendData },
+                        dataType:"json",
+                        cache: false,
+                        success: function(data) {
+                            if(data.error) {
+                                if (data.error.includes("Zbyt wiele zgłoszeń. Spróbuj ponownie za godzinę")) {
+                                    showError(tooManyReportError);
+                                    return false;
+                                } else {
+                                    isError = true;
+                                }
+                            } else if(data.success.includes("1")) {
+                                location.reload();
```

Comment:
Dlaczego przeładowujemy stronę? Jeśli już korzystamy z ajaxa do wysłania formularza, to może lepiej w tle pobrać treść nowej strony i umieścić ją w DOM - tym bardziej jeśli zmianie ulega fragment, a nie cały dokument?